components/RecentlyPlayed.tsx

```diff
@@ -0,0 +1,846 @@
+            renderShowGroup(showPlaylist, index)
+          )}
+        </View>
+      );
+    }
+
+    // Add loading indicator if loading more
+    if (loadingMore) {
+      content.push(
+        <View key="loading-more" style={styles.loadingMoreContainer}>
+          <ActivityIndicator size="small" color="#FFFFFF" />
+          <Text style={styles.loadingMoreText}>Loading previous show...</Text>
+        </View>
+      );
+    }
+
+    // Add end-of-day message if we've reached the end
+    if (hasReachedEndOfDay && !loadingMore) {
+      content.push(
+        <View key="end-of-day" style={styles.endOfDayContainer}>
+          <Text style={styles.endOfDayText}>No more shows for today</Text>
+            <TouchableOpacity onPress={() => navigation.navigate('Schedule')} style={styles.scheduleButton}>
+              <Text style={styles.scheduleButtonText}>View Full Schedule</Text>
+            </TouchableOpacity>
+        </View>
+      );
+    }
+
+    return content;
+  };
+
+  return (
+    <>
+      {/* Content */}
+          <ScrollView
+            ref={scrollViewRef}
+            style={styles.scrollView}
+            showsVerticalScrollIndicator={false}
+            bounces={true}
+            onScroll={handleScroll}
+            scrollEventThrottle={400}
+            refreshControl={
+              <RefreshControl
+                refreshing={refreshing}
+                onRefresh={handleRefresh}
+                tintColor="#FFFFFF"
+                colors={['#00843D', '#FFFFFF']}
+                progressBackgroundColor="#000000"
+                titleColor="#FFFFFF"
+                title=""
+              />
+            }
+          >
+            {/* Current Show Header - only show when there's a single show with songs */}
+            {currentShow && currentShow !== 'WMBR 88.1 FM' && showPlaylists.length === 1 && showPlaylists[0].songs.length > 0 && (
+              <View style={styles.currentShowHeader}>
+                <Text style={styles.currentShowTitle}>{currentShow}</Text>
+                <Text style={styles.currentShowSubtitle}>Now Playing</Text>
+              </View>
+            )}
+
+            {loading ? (
+              <View style={styles.loadingContainer}>
+                <ActivityIndicator size="large" color="#FFFFFF" />
+                <Text style={styles.loadingText}>Loading playlist...</Text>
+              </View>
+            ) : error ? (
+              <View style={styles.errorContainer}>
+                <Text style={styles.errorText}>{error}</Text>
+                <TouchableOpacity onPress={handleRefresh} style={styles.retryButton}>
+                  <Text style={styles.retryButtonText}>Retry</Text>
+                </TouchableOpacity>
+              </View>
+            ) : !currentShow || currentShow === 'WMBR 88.1 FM' ? (
+              <View style={styles.emptyContainer}>
+                <Text style={styles.emptyText}>No playlists found</Text>
```

Comment:
Reworded from "Pull up when a show is playing to see the playlist" since RecentlyPlayedDrawer prompts refresh/pull up on its own